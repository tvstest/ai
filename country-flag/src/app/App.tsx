import React from 'react'
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom'
import CountrySearch from 'app/pages/CountrySearch'
import CountryList from 'app/pages/CountryList'
import Navbar from 'app/components/Navbar'
import { Routes } from 'app/utils/enums/routes'
import { ThemeProvider } from '@mui/material'
import appTheme from 'app/configs/theme'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={appTheme}>
        <Navbar />
        <Switch>
          <Route exact path={Routes.Home} component={CountrySearch} />
          <Route exact path={Routes.CountriesByName} component={CountryList} />
          <Redirect to={Routes.Home} />
        </Switch>
      </ThemeProvider>
    </BrowserRouter>
  )
}
export default App
