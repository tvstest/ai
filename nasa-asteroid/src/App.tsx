import { ThemeProvider } from '@mui/material/styles'
import Navbar from 'app/components/Navbar'
import Routes from 'app/components/Routes'
import { Suspense } from 'react'
import theme from './app/config/theme'

const App: React.FC = () => {
  return (
    <Suspense fallback="Loading ...">
      <Navbar />
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
