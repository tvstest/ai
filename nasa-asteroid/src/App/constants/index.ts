export const NASA_BASE_URL = 'https://api.nasa.gov/neo/rest/v1/neo/'
export const NASA_API_KEY_PREFIX = `?api_key=${process.env.REACT_APP_NASA_API_KEY}`
