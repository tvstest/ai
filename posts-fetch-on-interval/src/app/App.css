.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* CSS Loader
--------------------------------------------------------------------------------------------------------*/
.loader-overlay {
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  background: rgba(215, 215, 215, 0.35);
  z-index: 9999;
}
.cssload-fond {
  margin: auto;
  position: fixed;
  z-index: 0.5;
  top: 50%;
  left: 50%;
  margin: -20px 0 0 -20px;
}

.loaderHide {
  display: none;
}

.loaderShow {
  display: block;
}
