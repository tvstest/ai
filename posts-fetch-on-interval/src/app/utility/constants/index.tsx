export const INTERNAL_SERVER_ERROR = 'Internal Server Error'
export const SOMETHING_WENT_WRONG = 'Something Went Wrong.'
export const ALL_DATA_FETCHED = 'All posts have been fetched.'
export const COULD_NOT_LOAD_DATA = 'Could not load list.'
export const TIMER_TEST_SECONDS = 21000
export const TIMER_MOCK_TIMEOUT = 40000
