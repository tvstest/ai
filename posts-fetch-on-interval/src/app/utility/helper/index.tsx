export const showLoader = (): void => {
  const loaderDiv = document.getElementById('loaderForAPICall')
  if (loaderDiv) {
    loaderDiv.classList.add('loaderShow')
    loaderDiv.classList.remove('loaderHide')
  }
}

export const hideLoader = (): void => {
  const loaderDiv = document.getElementById('loaderForAPICall')
  if (loaderDiv) {
    loaderDiv.classList.remove('loaderShow')
    loaderDiv.classList.add('loaderHide')
  }
}

export const sleep = (m: number): Promise<NodeJS.Timeout> =>
  new Promise((r) => setTimeout(r, m))
