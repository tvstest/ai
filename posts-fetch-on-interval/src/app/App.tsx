import React, { Suspense } from 'react'
import { ThemeProvider } from '@mui/material/styles'
import { ToastContainer } from 'react-toastify'
import Loader from './components/Loader'
import Navbar from './components/Navbar'
import { theme } from './app.theme'
import 'react-toastify/dist/ReactToastify.css'
import './App.css'
import PostsData from './components/PostsData'

function App() {
  return (
    <Suspense fallback={<Loader />}>
      <ThemeProvider theme={theme}>
        <ToastContainer />
        <Navbar />
        <PostsData />
        <Loader />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
