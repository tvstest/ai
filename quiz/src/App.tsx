import './App.css'
import { CircularProgress, ThemeProvider } from '@mui/material'
import { Suspense } from 'react'
import Routes from 'components/Routes'
import theme from 'configs/app-theme'
import { BrowserRouter as Router } from 'react-router-dom'

const App: React.FC = () => {
  return (
    <Suspense fallback={<CircularProgress />}>
      <ThemeProvider theme={theme}>
        <Router>
          <Routes />
        </Router>
      </ThemeProvider>
    </Suspense>
  )
}

export default App
