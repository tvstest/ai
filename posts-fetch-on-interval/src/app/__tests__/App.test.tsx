import * as React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { setupIntersectionObserverMock } from 'app/utility/testUtilities'
import App from '../App'

beforeEach(() => {
  setupIntersectionObserverMock()
})

test('renders posts text in navbar', () => {
  render(<App />)
  const linkElement = screen.getByText(/posts/i)
  expect(linkElement).toBeInTheDocument()
})

test('renders App correctly', () => {
  const domTree = renderer.create(<App />).toJSON()
  expect(domTree).toMatchSnapshot()
})
