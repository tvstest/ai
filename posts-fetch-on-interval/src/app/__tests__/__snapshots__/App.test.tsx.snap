// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
Array [
  <div
    className="Toastify"
  />,
  <div
    className="makeStyles-root-22"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1xwc9t2-MuiToolbar-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap makeStyles-title-24 css-1sr3gzp-MuiTypography-root"
        >
          Posts
        </h6>
      </div>
    </header>
  </div>,
  <div
    className="makeStyles-root-25 MuiBox-root css-0"
  >
    <div
      className="makeStyles-postListContainer-32 MuiBox-root css-0"
    >
      <div
        className="makeStyles-searchContainer-31 MuiBox-root css-0"
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-searchInput-33 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pw3wan-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="search"
            id="search-label"
          >
            Search post
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1q5oif2-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="search"
              disabled={false}
              id="search"
              name="search"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-173wfxe"
              >
                <span>
                  Search post
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="makeStyles-searchContainer-31 MuiBox-root css-0"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-fullWidth css-eozd4h-MuiAutocomplete-root"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-autocomplete-35 css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pw3wan-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              id="undefined-label"
            >
              Title
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1g6xlmm-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1j5luls-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="search"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-173wfxe"
                >
                  <span>
                    Title
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textField-36 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1rzclia-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor="date"
            id="date-label"
          >
            Created At
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1q5oif2-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              defaultValue=""
              disabled={false}
              id="date"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="date"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1z7n62"
              >
                <span>
                  Created At
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-0"
      >
        <div
          className="makeStyles-noData-34"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-10zdtnx-MuiTypography-root"
          >
             No data found 
          </p>
        </div>
      </div>
      <div
        style={
          Object {
            "opacity": "0",
          }
        }
      >
        End of data
      </div>
    </div>
  </div>,
  <div
    className="loaderHide"
    id="loaderForAPICall"
  >
    <div
      className="cssload-fond"
    >
      <div
        className="makeStyles-root-42"
      >
        <span
          className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
    <div>
      <div
        className="loader-overlay-nav"
      />
      <div
        className="loader-overlay"
        id="loaderNav"
      />
    </div>
  </div>,
]
`;
